functions/array/array_rand.js

```diff
@@ -1,38 +1,32 @@
-function array_rand (input, num_req) {
-  //  discuss at: http://phpjs.org/functions/array_rand/
-  // original by: Waldo Malqui Silva (http://waldo.malqui.info)
-  //   example 1: array_rand( ['Kevin'], 1 );
-  //   returns 1: 0
+function array_rand(array, num) {
+  //       discuss at: http://phpjs.org/functions/array_rand/
+  //      original by: Waldo Malqui Silva (http://waldo.malqui.info)
+  // reimplemented by: Rafał Kukawski
+  //        example 1: array_rand( ['Kevin'], 1 );
+  //        returns 1: 0
 
-  var indexes = []
-  var ticks = num_req || 1
-  var checkDuplicate = function (input, value) {
-    var exist = false,
-      index = 0,
-      il = input.length
-    while (index < il) {
-      if (input[index] === value) {
-        exist = true
-        break
-      }
-      index++
-    }
-    return exist
-  }
+  // By using Object.keys we support both, arrays and objects
+  // which phpjs wants to support
+  var keys = Object.keys(array);
```

Comment:
I made this patch as a pull request to be able to discuss/do code review.
Object.keys returns all keys as strings. Do we want to convert integer keys to number or is the current result OK?